crates/client/rpc-core/Cargo.toml

```diff
@@ -28,3 +28,5 @@ sp-blockchain = { workspace = true, default-features = true }
 starknet_api = { workspace = true, default-features = false }
 frame-support = { workspace = true }
 sp-core = { workspace = true }
+starknet-providers = { git = "https://github.com/xJonathanLEI/starknet-rs", branch = "dev/jsonrpc_0_3_0", default-features = false }
```

Comment:
would be better to use `workspace = true` here to inherit the version defined in the workspace Cargo toml.